### DISCUSSION TITLE
Mail configuration

### DISCUSSION THREAD

**Discussion by nnachtegael** [immense/Remotely#206] [Q&A]
Hi,
Can someone explain me how to configure the mail.
I set all my infos in the mail section of the account/server config but the save & test gives me always a 500 timeout error
My remotely is installed alone on a Ubuntu 20.04 vps.
I installed the mailutils and ssmtp packets. 
Thx

**Comment by stevesobol** [comment]
On 2021-01-13 11:27, nnachtegael wrote:

> Hi,
> Can someone explain me how to configure the mail.
> I set all my infos in the mail section of the account/server config but the save & test gives me always a 500 timeout error
> My remotely is installed alone on a Ubuntu 20.04 vps.
> I installed the mailutils and ssmtp packets.
> Thx

Which port are you using to send mail? 25? 587?

Port 25 is blocked by many virtual server providers and ISPs.  

587 is not. 587 is meant to be used for authenticated SMTP only, meaning
your mail server should be set to require valid account credentials.
That's how my server is set up... 25 for incoming messages from other
mail servers; 587 for people sending mail out (or programs that need to
send mail, in this case), and I require STARTTLS. 

My guess is that this is PROBABLY not a Remotely issue.

**Comment by nnachtegael** [comment]
OK it's working with
smtp.gmail.com
ssl on
port 587

AND
sudo ufw allow 587

😁

**Reply by nnachtegael** [reply]
Thx

**Comment by ittechdb** [comment]
Have server configured. Still can't get emails to send from Remotely.. Installed mailutils and ssmtp also.. Can send from ssmtp, but still get 504 from Remotely Save and Test.. Any ideas or screenshots available? Thx!!!!

**Comment by stevesobol** [comment]
@ittechdb details, please! :)

**Reply by ittechdb** [reply]
Hello! Have the ubuntu server up and running. Remotely is running.
I am trying to get the email to work, but I get the 504 timeout when testing.
Am I missing anything? Even spun up a 18.04 version of Ubuntu to make sure it was the same result.

**Reply by stevesobol** [reply]
I can't answer that without knowing a little about your setup. I'd refer you back to my first reply, to nnachteagael, for some pointers. Server configuration varies from provider to provider, so I do actually need details...

I'm pretty sure I can help you to resolve your problem. But I need your assistance first. :)

**Reply by ittechdb** [reply]
Ahh. No I am not running the server with a provider. I am running it locally. Using gmail smtp.
What info are you looking for? Fresh install 18.04, Remotely works besides email. 504 Gateway TIme-out when you click Save and Test.
The gmail smtp info I entered is correct. port 465, SSL selected, etc..

**Reply by stevesobol** [reply]
THAT was the detail I needed. You're using Gmail SMTP.

@nnachtegael is also using Gmail SMTP, so you want to refer back to **his** reply to **me.**

Google documentation is [here](https://support.google.com/a/answer/176600?hl=en), if you need it.

**Reply by ittechdb** [reply]
Ok thxxx!!! I will check it out.

**Comment by houstonbofh** [comment]
I too was having this issue on Ubuntu 18.04.  The mail server is a standard Linux MTA.  I was using ssl on port 465, and decided to try 587 in spite of that being TLS not SSL and it worked.  Not sure why as SSL on 465 does work with that email server.  The logs were less than helpful being only;
`
Error | 4/5/2021 2:08:54 PM -05:00 | The operation has timed out. | System.Net.Mail | at  System.Net.Mail.SmtpClient.Send(MailMessage message)    at Remotely.Server.Services.EmailSenderEx.SendEmailAsync(String  email, String replyTo, String subject, String htmlMessage, String  organizationID) in D:\a\1\s\Server\Services\EmailSender.cs:line 49

`
I recommend changing the field in the config page to SSL/TLS, and figuring out why SSL is silently crashing.

**Comment by stevesobol** [comment]
On 2021-04-05 12:30, Lee Sharp wrote:

> I too was having this issue on Ubuntu 18.04. The mail server is a standard Linux MTA. I was using ssl on port 465, and decided to try 587 in spite of that being TLS not SSL and it worked. Not sure why as SSL on 465 does work with that email server. The logs were less than helpful being only;
> `
> Error | 4/5/2021 2:08:54 PM -05:00 | The operation has timed out. | System.Net.Mail | at System.Net.Mail.SmtpClient.Send(MailMessage message) at Remotely.Server.Services.EmailSenderEx.SendEmailAsync(String email, String replyTo, String subject, String htmlMessage, String organizationID) in D:\a\1\s\Server\Services\EmailSender.cs:line 49 
> 
> `
> I recommend changing the field in the config page to SSL/TLS, and figuring out why SSL is silently crashing. 
> 
> --
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub [1], or unsubscribe [2].

SSL v2 and V3 are deprecated, anyhow. TLS is more secure, and TLS 1.2 is
recommended over 1.0 and 1.1. 

Links:
------
[1]
https://github.com/lucent-sea/Remotely/discussions/206#discussioncomment-571620
[2]

**Reply by bitbound** [reply]
Yeah, the term "SSL" is just legacy verbiage.  It doesn't actually use SSL anymore.  In fact, it looks like the .NET Core SmtpClient will throw an error if StartTls isn't supported. 

https://github.com/dotnet/corefx/blob/master/src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs#L223

I'll change the language on the config page.  :)

**Reply by houstonbofh** [reply]
I would even add a note to the config page that SSL is no longer supported and to use TLS and 587.  The fact that is said SSL and not TLS was why I have a sore head now. :)  Thanks for the quick fix.

**Reply by kpdillon** [reply]
So what is the fix exactly?  I am trying 587 with smtp.gmail.com and I am getting the following error.

at System.Net.Mail.MailCommand.CheckResponse(SmtpStatusCode statusCode, String response) at System.Net.Mail.MailCommand.Send(SmtpConnection conn, Byte[] command, MailAddress from, Boolean allowUnicode) at System.Net.Mail.SmtpTransport.SendMail(MailAddress sender, MailAddressCollection recipients, String deliveryNotify, Boolean allowUnicode, SmtpFailedRecipientException& exception) at System.Net.Mail.SmtpClient.Send(MailMessage message) at Remotely.Server.Services.EmailSenderEx.SendEmailAsync(String email, String replyTo, String subject, String htmlMessage, String organizationID) in D:\a\1\s\Server\Services\EmailSender.cs:line 49

**Reply by houstonbofh** [reply]
Google is picky.  Did you do all of this?  https://support.google.com/a/answer/176600?hl=en

**Reply by kpdillon** [reply]
Hi @houstonbofh Yes, and I relay other apps through google just fine so I am stumped why this is not working.  It is a Digital Ocean VPS running Ubuntu 18

**Reply by houstonbofh** [reply]
Not sure why unless you have different settings.  Your smtp username and your email address should match, for example.  It could also be Google setting limits.  Try creating a new gmail account and using that.

**Reply by bitbound** [reply]
It looks like you need to create an app password if you have two-factor authentication enabled.  It won't work with your account password.  I tested it with my personal Gmail account, and it's working.  (The app password doesn't contain spaces even though it looks like it does.)

![image](https://user-images.githubusercontent.com/20995508/114326691-d6b2d100-9aea-11eb-87c2-bfda1186276c.png)

**Reply by bitbound** [reply]
I could also imagine situations where an email provider or other service blocks certain IP ranges due to repeated abuse (e.g. people setting up spam bots from VPS's).

**Reply by kpdillon** [reply]
My VPS is not blocked, I have tried different gmail accounts.  I can telnet to port 587 from the VPS no problem. I have less secured apps turned on.  I am truly puzzled as to why it doesn't work for me.

**Reply by houstonbofh** [reply]
I am stumped as well.  Are you early enough in that you could try a fresh install just to make sure it is not leftover config stuff from trying other things?  Or post your config file here...  Unfortunately what you would have to blur for privacy is what we probably need to see.

**Reply by kpdillon** [reply]
Yes, I could reinstall. No easy way to migrate to a new server yet?  Are there specific logs on the VPS that would provide more info into why this is failing?

**Reply by bitbound** [reply]
I see you pasted the stack trace, but there should also be an error message in the logs too.  What does that part say?

**Reply by bitbound** [reply]
And by logs, I mean the logs in the Remotely web interface on the Server Logs tab.

**Reply by kpdillon** [reply]
The SMTP server requires a secure connection or the client was not authenticated. The server response was: 5.7.0 Authentication Required. Learn more at.....

**Reply by bitbound** [reply]
That's the message when two-factor is preventing login. I'm pretty sure you'll need to create an app password as described above.

I know you mentioned other apps, but they're probably using some OAuth flow to complete the two-factor. The .NET SmtpClient doesn't support that.

This is a good thing. It means two-factor is doing its job.

**Reply by kpdillon** [reply]
I have both a gmail.com and g-suite account that I've tried.  Neither has 2FA turned on.  The gmail.com account doesn't allow me to create an app password.  I do get notifications that Google has prevented a login attempt when I enter in the wrong password so I know it's getting to the relay I just get stuck on this.  If I can send you a private message I am happy to give you access to my VPS if it would help this project.

**Reply by bitbound** [reply]
I replaced the built-in SmtpClient with MailKit.  Might help your issue.  https://github.com/lucent-sea/Remotely/issues/291
